### PR TITLE
feat: pre validate service

### PR DESCRIPTION
**Pull request by AkaShark**
closed https://github.com/tisfeng/Easydict/issues/714

### CODE REVIEW

**Comment by CanglongCl** [comment]
@AkaShark Hi I made some changes to your branch. Now while user click the toggle, it will be turned into a `ProgressView` individually. Error message alert is also implemented.

**Comment by AkaShark** [comment]
> @AkaShark Hi I made some changes to your branch. Now while user click the toggle, it will be turned into a `ProgressView` individually. Error message alert is also implemented.

Thank you for your commit! As someone new to SwiftUI, your code helps me learn more, and it improves this feature significantly.

File: Easydict/App/Localizable.xcstrings [hunk 1]
```diff
@@ -1964,6 +1964,17 @@
         }
       }
     },
+    "fail_validate_service %@" : {
+      "extractionState" : "stale",
```
Comment (tisfeng) [inline Easydict/App/Localizable.xcstrings:1968]: If this string is not used, please remove it.

File: Easydict/Swift/View/SettingView/Tabs/TabView/ServiceTab.swift [hunk 2]
```diff
@@ -170,17 +171,87 @@ private class ServiceItemViewModel: ObservableObject {
...
+                    )
+                    return
+                }
+
+                // service enabel open the switch and toggle enable status
+                self.service.enabled = true
+                self.service.enabledQuery = true
+                EZLocalStorage.shared().setService(self.service, windowType: self.viewModel.windowType)
+                self.viewModel.postUpdateServiceNotification()
+//                self.objectWillChange.send()
```
Comment (tisfeng) [inline Easydict/Swift/View/SettingView/Tabs/TabView/ServiceTab.swift:217]: Can this code be removed?

**Comment by tisfeng** [comment]
I improved the UI, aligning the ProgressView and Toggle 0b51d9ad

## Before
image

## Now 
image

**Review by tisfeng — CHANGES_REQUESTED** [review]
For the new localizations, we need to add translations in English, Simplified Chinese, Traditional Chinese, and Slovak.

We just need to ensure that the translations in Chinese and English are correct. As for other languages, such as Slovak, translation tools can be used. 

I usually use Cursor for auto-completion, and later there will be users familiar with that language to proofread https://github.com/tisfeng/Easydict/pull/708#issuecomment-2447086972

image


image

**Comment by tisfeng** [comment]
I'm considering removing the en-CA localization because it's almost the same as en, and we're not familiar with it. It's not worth spending time to differentiate between them. 

Initially, we added en-CA just to provide users with a localization example https://github.com/tisfeng/Easydict/pull/231. Moreover, there's now a better Slovak localization example https://github.com/tisfeng/Easydict/pull/668.

**Comment by AkaShark** [comment]
> For the new localizations, we need to add translations in English, Simplified Chinese, Traditional Chinese, and Slovak.对于新的本地化，我们需要添加英语、简体中文、繁体中文和斯洛伐克语的翻译。
> 
> We just need to ensure that the translations in Chinese and English are correct. As for other languages, such as Slovak, translation tools can be used.我们只需要确保中文和英文的翻译是正确的。至于其他语言，比如斯洛伐克语，可以使用翻译工具。
> 
> I usually use Cursor for auto-completion, and later there will be users familiar with that language to proofread [#708 (comment)](https://github.com/tisfeng/Easydict/pull/708#issuecomment-2447086972)我一般都是用Cursor来做自动补全，后面会有熟悉那个语言的用户来校对 [#708 (comment)](https://github.com/tisfeng/Easydict/pull/708#issuecomment-2447086972)
> 
> image image

That seems like a good way.

**Review by tisfeng — APPROVED** [review]
Looks good.

**Comment by Jerry23011** [comment]
> I'm considering removing the en-CA localization because it's almost the same as en, and we're not familiar with it. It's not worth spending time to differentiate between them.
> 
> Initially, we added en-CA just to provide users with a localization example #231. Moreover, there's now a better Slovak localization example #668.

+1 for this

**Pull request merged**